### ISSUE TITLE
Limitation on number of vectors that can be searched for dot product similarity

### ISSUE THREAD

**Issue by ksrinivs64** [mikeheddes/dothash#1]
Hi thanks for a very nice paper and the code - does the dothash solution scale to millions of vectors (say millions of documents for vector search)?  Or is it currently limited by whatever can be computed in memory?  Thanks

**Comment by mikeheddes** [comment]
Hi, thank you for your interest in DotHash. DotHash simply requires storing a vector for each document or node. You could use something like a vector database to scale this to millions of vectors.

**Comment by ksrinivs64** [comment]
Yes, but I was looking at ANNs and most claim to do very poorly with high dimensional vectors.  Have you tried any particular one?  Would you recommend something if you have?  Thanks again - very cool work.

**Comment by mikeheddes** [comment]
I have not used vector databases, the experiments we did were small enough that everything fits in memory. Could you elaborate on the following:
> I was looking at ANNs and most claim to do very poorly with high dimensional vectors

It is not clear to me what performs poorly, the ANN? Or the vector database?

**Comment by ksrinivs64** [comment]
As far as I know all vector databases scale by space partitioning
algorithms and the ones I looked at like FAISS said they become really
inaccurate with high dimensional vectors.
Kavitha
